src/Authentication.Abstractions/AzureEnvironment.cs

```diff
@@ -314,6 +188,37 @@ private static AzureEnvironment MapArmToAzureEnvironment(ArmMetadata armMetadata
                 azureEnvironment.ServiceManagementUrl += "/";
             }
 
+            if (!string.IsNullOrEmpty(armMetadata.MicrosoftGraphResourceId))
+            {
+                azureEnvironment.SetProperty(ExtendedEndpoint.MicrosoftGraphEndpointResourceId, armMetadata.MicrosoftGraphResourceId);
+                //Default ARM endpoint doens't provide MicrosoftGraphUrl. Keep it here just in case.
+                if (armMetadata.MicrosoftGraphResourceId.EndsWith("/"))
+                {
+                    azureEnvironment.SetProperty(ExtendedEndpoint.MicrosoftGraphUrl,
+                        armMetadata.MicrosoftGraphResourceId.TrimEnd('/'));
+                }
+            }
+
+            if (!string.IsNullOrEmpty(armMetadata.AttestationResourceId))
+            {
+                azureEnvironment.SetProperty(ExtendedEndpoint.AzureAttestationServiceEndpointResourceId, armMetadata.AttestationResourceId);
+                if (!string.IsNullOrEmpty(armMetadata.Suffixes.AttestationEndpoint))
+                {
+                    azureEnvironment.SetProperty(ExtendedEndpoint.AzureAttestationServiceEndpointSuffix, armMetadata.Suffixes.AttestationEndpoint);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(armMetadata.SynapseAnalyticsResourceId))
+            {
+                azureEnvironment.SetProperty(ExtendedEndpoint.AzureSynapseAnalyticsEndpointSuffix, armMetadata.SynapseAnalyticsResourceId);
+                if (!string.IsNullOrEmpty(armMetadata.Suffixes.SynapseAnalytics))
+                {
+                    azureEnvironment.SetProperty(ExtendedEndpoint.AzureSynapseAnalyticsEndpointSuffix, armMetadata.Suffixes.SynapseAnalytics);
+                }
+            }
+            //Todo:MhsmDns=ManagedHsmServiceEndpointSuffix;AzureManagedHsmServiceEndpointResourceId not provided
+            //Todo:AcrLoginServer=ContainerRegistryEndpointSuffix;ContainerRegistryEndpointResourceId not provided
```

Comment:
Can you explain more (in the code) about these todos?
What do we need to do and when can we do it?

Reply:
AzureManagedHsmServiceEndpointResourceId and ManagedHsmServiceEndpointSuffix are currently hard-coded in our product. and Now ManagedHsmServiceEndpointSuffix(MhsmDns) is provided by ArmMetadata. But it doesn't' make sense to just refresh ManagedHsmServiceEndpointSuffix by MhsmDns from ARM. If we want to refresh AzureManagedHsmServiceEndpointResourceId with reference to ManagedHsmServiceEndpointSuffix,  it may require the process of verification as something we do for MicrosoftGraph endpoints.

I think we can do it when ManagedHsmService endpoints and ContainerRegistry endpoints are also required from service team.

Reply:
It makes sense. Please put it in code as comments, so we'll understand the context when revist it in the future.